WooCommerce/Classes/POS/Controllers/PointOfSaleCouponsController.swift

```diff
@@ -64,20 +66,47 @@ import protocol Yosemite.PointOfSaleCouponServiceProtocol
 
 @available(iOS 17.0, *)
 private extension PointOfSaleCouponsController {
+    /// Loads the first page by attempting to load the first page from local storage
+    /// then syncs from the remote regardless the result
+    func loadFirstPage() async {
+        do {
+            let storedCoupons = try await couponProvider.provideLocalPointOfSaleCoupons()
+
+            if !storedCoupons.isEmpty {
+                setCouponsLoadedViewState(storedCoupons, hasMoreItems: true)
+            }
+            let coupons = try await couponProvider.providePointOfSaleCoupons(pageNumber: 1)
+            if !coupons.items.isEmpty {
+                setCouponsLoadedViewState(coupons.items, hasMoreItems: true)
```

Comment:
Should this be `coupons.hasMorePages` instead of true?

```suggestion
                setCouponsLoadedViewState(coupons.items, hasMoreItems: coupons.hasMorePages)
```

Reply:
Not in this case, it's a bit clearer after merging trunk: 
```
let storedCoupons = try await couponProvider.provideLocalPointOfSaleCoupons()
if !storedCoupons.isEmpty {
  setCouponsLoadedViewState(storedCoupons, hasMoreItems: true)
}
```

Since we fetch the first batch from local storage, we receive `[Coupon]`, which doesn't have a value for hasMorePages, we just assume there will be a next page until we can confirm it via the remote and paged coupons.